test/integration/src/test/java/io/pravega/test/integration/ReaderGroupNotificationTest.java

```diff
@@ -189,6 +196,87 @@ public void testSegmentNotifications() throws Exception {
         assertEquals(2, segmentNotificationPostScale.getNumOfSegments());
     }
 
+    @Test(timeout = 40000)
+    public void testTargetRateNotifications() throws Exception {
+        final String streamName = "stream1";
+        StreamConfiguration config = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.byEventRate(10, 2, 3))
+                .build();
+        Controller controller = controllerWrapper.getController();
+        controllerWrapper.getControllerService().createScope(SCOPE).get();
+        controller.createStream(SCOPE, streamName, config).get();
+        @Cleanup
+        ConnectionFactory connectionFactory = new ConnectionFactoryImpl(ClientConfig.builder()
+                .controllerURI(URI.create("tcp://localhost"))
+                .build());
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE, controller, connectionFactory);
+        EventStreamWriter<String> writer1 = clientFactory.createEventWriter(streamName, new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+        List<String> routingKeys = new ArrayList<>();
+        routingKeys.add("0");
+        @Cleanup
+        ReaderGroupManager groupManager = new ReaderGroupManagerImpl(SCOPE, controller, clientFactory,
+                connectionFactory);
+        groupManager.createReaderGroup("reader", ReaderGroupConfig
+                .builder().disableAutomaticCheckpoints().stream(Stream.of(SCOPE, streamName)).groupRefreshTimeMillis(0).build());
+        @Cleanup
+        ReaderGroup readerGroup = groupManager.getReaderGroup("reader");
+        @Cleanup
+        EventStreamReader<String> reader1 = clientFactory.createReader("readerId", "reader", new JavaSerializer<>(),
+                ReaderConfig.builder().initialAllocationDelay(0).build());
+
+        val notificationResults = new ArrayBlockingQueue<SegmentNotification>(10);
+
+        //Add segment event listener
+        Listener<SegmentNotification> l1 = notification -> {
+            log.info("Number of Segments: {}, Number of Readers: {}", notification.getNumOfSegments(), notification.getNumOfReaders());
+            notificationResults.add(notification);
+        };
+        SegmentNotifier segmentNotifier = (SegmentNotifier) readerGroup.getSegmentNotifier(executor);
+        segmentNotifier.registerListener(l1);
+
+        long start = System.currentTimeMillis();
+        CompletableFuture.runAsync(() -> {
+            while (System.currentTimeMillis() - start < Duration.ofMinutes(3).toMillis()) {
```

Comment:
So you're writing an integration test that deliberately takes 3 minutes to run? The only thing this will do is make our build take longer for everyone who wants to run it.